cloudify_nsx/library/nsx_esg_dlr.py

```diff
@@ -995,15 +1071,233 @@ def update_common_edges(client_session, resource_id, kwargs, esg_restriction):
+                                    "INFO",
+                                    "DEBUG"
+                                ]
+                            },
+                            "enable": {
+                                "default": False,
+                                "type": "boolean"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    else:
+        validation_rules_routing = {
+            "enabled": {
+                "default": True,
+                "type": "boolean"
+            },
+            "staticRouting": {
+                "set_none": True,
+                "sub": {
+                    "defaultRoute": {
+                        "set_none": True,
+                        "sub": {
+                            "gatewayAddress": {
+                                "set_none": True
+                            },
+                            "vnic": {
+                                "set_none": True
+                            },
+                            "mtu": {
+                                "set_none": True
+                            }
+                        }
+                    }
+                }
+            },
+            "routingGlobalConfig": {
+                "sub": {
+                    "routerId": {
+                        "set_none": True
+                    },
+                    "ecmp": {
+                        "default": False,
+                        "type": "boolean"
+                    },
+                    "logging": {
+                        "sub": {
+                            "logLevel": {
+                                "default": "INFO",
+                                "caseinsensitive": True,
+                                "values": [
+                                    "EMERGENCY",
+                                    "ALERT",
+                                    "CRITICAL",
+                                    "ERROR",
+                                    "WARNING",
+                                    "NOTICE",
+                                    "INFO",
+                                    "DEBUG"
+                                ]
+                            },
+                            "enable": {
+                                "default": False,
+                                "type": "boolean"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+    _, routing = common.get_properties_and_validate(
+        'routing', kwargs, validation_rules_routing
+    )
     routing_global_config(
         client_session, resource_id,
         routing['enabled'], routing['routingGlobalConfig'],
         routing['staticRouting']
     )
 
-    _, ospf = common.get_properties_and_validate('ospf', kwargs)
-    _, bgp = common.get_properties_and_validate('bgp', kwargs)
+    if esg_restriction:
```

Comment:
It might be better to write the basic validation rules and then add the extra elements to them if esg_restriction? That way, if the basics need to be changed they only need changing once.